### PR TITLE
UA : clean streams

### PR DESCRIPTION
**Pull request by BellezaEmporium**
- Adaptive links
- Prevent certificate issue for some Cosmonova channels, changing link and adaptive.

### CODE REVIEW

File: streams/ua.m3u [hunk 1]
```diff
@@ -1,19 +1,13 @@
 #EXTM3U
 #EXTINF:-1 tvg-id="1Plus1Sport.ua" status="online",1+1 Спорт (720p) [Not 24/7]
 https://live-k2301-kbp.1plus1.video/sport/smil:sport.smil/playlist.m3u8
-#EXTINF:-1 tvg-id="" status="online",4 канал (576p)
-http://95.67.106.10/hls/nta_ua_low/index.m3u8
 #EXTINF:-1 tvg-id="" status="online",4 канал (720p)
-http://95.67.106.10/hls/nta_ua_mid/index.m3u8
-#EXTINF:-1 tvg-id="" status="online",4 канал (1080p)
-http://95.67.106.10/hls/nta_ua_hi/index.m3u8
+http://95.67.106.10/hls/nta_ua.m3u8
```
Comment (freearhey) [inline streams/ua.m3u:5]: New link is not working:
image

File: streams/ua.m3u [hunk 2]
```diff
@@ -89,14 +77,8 @@ https://rtsp.cdn.ua/tv5.zp.ua_live/tv5/playlist.m3u8
...
 https://cdn10.live-tv.od.ua:8083/odtrkod/odtrkod-abr/playlist.m3u8
-#EXTINF:-1 tvg-id="UATV.ua" status="online",UATV (576p)
-https://ua-tv-hls2.cosmonova.net.ua/hls/ua-tv_ua_low/index.m3u8
-#EXTINF:-1 tvg-id="UATV.ua" status="online",UATV (720p)
-http://95.67.106.242/hls/ua-tv_ua_mid/index.m3u8
 #EXTINF:-1 tvg-id="UATV.ua" status="online",UATV (720p)
-https://ua-tv-hls2.cosmonova.net.ua/hls/ua-tv_ua_mid/index.m3u8
-#EXTINF:-1 tvg-id="UATV.ua" status="online",UATV (720p)
-https://ua-tv-hls3.cosmonova.net.ua/hls/ua-tv_ua_mid/index.m3u8
+https://ua-tv.cdn-02.cosmonova.net.ua/hls/ua-tv_ua.m3u8
```
Comment (freearhey) [inline streams/ua.m3u:81]: image

File: streams/ua.m3u [hunk 3]
```diff
@@ -181,16 +159,8 @@ https://cdn1.live-tv.od.ua:8083/1tvod/1tvod/playlist.m3u8
...
-#EXTINF:-1 tvg-id="PervyyDelovoy.ua" status="online",Первый Деловой (720p)
-http://95.67.127.156/hls/pershij-dlovij_ua_hi/index.m3u8
-#EXTINF:-1 tvg-id="PervyyDelovoy.ua" status="online",Первый Деловой (720p)
-http://pershij-dlovij-hls1.cosmonova.net.ua/hls/pershij-dlovij_ua_hi/index.m3u8
-#EXTINF:-1 tvg-id="PershiyDiloviy.ua" status="online",Перший Діловий (576p)
-http://pershij-dlovij-hls3.cosmonova.net.ua/hls/pershij-dlovij_ua_mid/index.m3u8
-#EXTINF:-1 tvg-id="PershiyDiloviy.ua" status="online",Перший Діловий (720p)
-http://pershij-dlovij-hls3.cosmonova.net.ua/hls/pershij-dlovij_ua_hi/index.m3u8
+#EXTINF:-1 tvg-id="PervyyDelovoy.ua" status="online",Первый Деловой (720p) [Not 24/7]
+http://pershij-dlovij.cdn-02.cosmonova.net.ua/hls/pershij-dlovij_ua.m3u8
```
Comment (freearhey) [inline streams/ua.m3u:163]: image

**Review by freearhey — CHANGES_REQUESTED** [review]
Please check the marked links again.

File: streams/ua.m3u [hunk 4]
```diff
@@ -204,15 +174,9 @@ https://app.live.112.events/hls-ua/112hd_mid/index.m3u8
 #EXTINF:-1 tvg-id="Pershiynezalezhniy.ua" status="online",Перший незалежний (1080p) [Timeout]
 https://app.live.112.events/hls/112hd_hi/index.m3u8
 #EXTINF:-1 tvg-id="" status="online",ПравдаТУТ (720p)
-http://95.67.17.131/hls/pravdatytkievshina_ua_hi/index.m3u8
-#EXTINF:-1 tvg-id="" status="online",ПравдаТУТ (720p)
-http://pravdatytkievshina-hls2.cosmonova.net.ua/hls/pravdatytkievshina_ua_hi/index.m3u8
+http://pravdatytkievshina.cdn-02.cosmonova.net.ua/hls/pravdatytkievshina_ua.m3u8
```
[the remark this answers is not in the document]
Comment (freearhey) [reply streams/ua.m3u:177]: All I get is a 404 error. Tested with VLC and Elmedia Player.

image

[the remark this answers is not in the document]
Comment (LaneSh4d0w) [reply streams/ua.m3u:177]: Browser extension (Firefox) and SPlayer in Android load them perfectly. Maybe an user-agent issue.

File: streams/ua.m3u (hunk 1 above)
Comment (freearhey) [reply streams/ua.m3u:5]: I don't know what exactly the problem is, but when the old links work without problems 
image
the new one still return an error
image

Comment (LaneSh4d0w) [reply streams/ua.m3u:5]: Let me see if this could be an user-agent issue.

**Comment by LaneSh4d0w** [comment]
Warning : Ukraina 24's YouTube channel is streaming 5 different streams at once, the stream having most persons in is chosen first by YT's API.

File: streams/ua.m3u (hunk 1 above)
Comment (LaneSh4d0w) [reply streams/ua.m3u:6]: Just tested my Firefox user agent against the website with VLC, works.

Comment (freearhey) [reply streams/ua.m3u:6]: Unfortunately, I still have the same 404 in VLC.

Comment (LaneSh4d0w) [reply streams/ua.m3u:6]: This is really weird. I guess i'll have to keep both the adaptive and the other links then.

Comment (LaneSh4d0w) [reply streams/ua.m3u:6]: Added the original links back. Problem should be solved.

**Review by freearhey — DISMISSED** [review]

**Review by Dum4G — DISMISSED** [review]

**Review by freearhey — APPROVED** [review]

**Review by Dum4G — APPROVED** [review]

**Pull request merged**